File: src/main/js/locations/location_service.js
angular.module('action').service('LocationService', ['$http', function($http) {
    this.findAll = function() {
        console.log("Getting all locations");
        return $http.get('/api/locations');
    };

    this.get = function(id) {
        console.log("Getting location with id " + id);
        return $http.get('/api/locations/' + id);
    };

    this.saveLocation = function(location) {
        return $http.post('/api/locations', location);
    };
}]);

File: src/main/java/no/swact/action/services/LocationService.java
package no.swact.action.services;

import no.swact.action.models.Location;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface LocationService {

    List<Location> findAll();

    Location save(Location location);

    Location findOne(Long id);
}
